File: src/filler.ts
import { fetchRatesForDate } from './client.js';
import { db } from './db.js';

async function wait(ms: number) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

export async function fillMissingRates() {
  const latest = db.getLatestRates()[0]?.date || '2025'
  const startDate = new Date(latest);
  const endDate = new Date();
  const currentDate = new Date(startDate);
  console.log(startDate)

  while (currentDate <= endDate) {
    const dateStr = currentDate.toISOString().split('T')[0];
    await fetchRatesForDate(dateStr);
    currentDate.setDate(currentDate.getDate() + 1);

    await wait(5 * 1000);
  }
}

await fillMissingRates()

File: src/filler.py
import asyncio
from datetime import datetime, timedelta
from .db import db
from .client import fetch_rates_for_date

async def fill_missing_rates():
    # Get the latest date we have in the DB
    latest_rates = db.get_latest_rates()
    if not latest_rates:
        start_date_str = "2025-01-01" # Default start if empty
    else:
        start_date_str = latest_rates[0]['date']
    
    start_date = datetime.strptime(start_date_str, "%Y-%m-%d")
    end_date = datetime.now()
    current_date = start_date

    print(f"[Filler] Starting from {start_date_str}")

    while current_date <= end_date:
        date_str = current_date.strftime("%Y-%m-%d")
        if not db.has_data_for_date(date_str):
            print(f"[Filler] Fetching missing rates for {date_str}")
            await fetch_rates_for_date(date_str)
            # Throttle API calls
            await asyncio.sleep(5)
        
        current_date += timedelta(days=1)

if __name__ == "__main__":
    asyncio.run(fill_missing_rates())
